mystore/app/models/spree/promotion/actions/create_product_adjustments.rb

```diff
@@ -0,0 +1,40 @@
+module Spree
+  class Promotion
+    module Actions
+      class CreateProductAdjustments < PromotionAction
+        include Spree::CalculatedAdjustments
+        include Spree::AdjustmentSource
+
+        has_many :promotion_action_products, foreign_key: :promotion_action_id
+        has_many :products, through: :promotion_action_products
+
+        validates :products, presence: true, on: :update
+
+        before_validation -> { self.calculator ||= Calculator::PercentOnLineItem.new }
+
+        def perform(options = {})
+          order = options[:order]
+          promotion = options[:promotion]
+          eligible_line_items = order.line_items.joins(:product).where(spree_products: { id: product_ids })
+          create_unique_adjustments(order, eligible_line_items) do |line_item|
+            promotion.line_item_actionable?(order, line_item)
+          end
+        end
+
+        def compute_amount(line_item)
+          return 0 unless promotion.line_item_actionable?(line_item.order, line_item)
```

Comment:
@rajneeshsharma9 Check if need to add the `eligible_line_items` condition here as well